test_scripts/Polices/user_consent_of_Policies/ATF_isPermissionsConsentNeeded_true.lua

```diff
@@ -0,0 +1,139 @@
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonSteps:DeleteLogsFileAndPolicyTable()
+
+function Test:Unregister_default_app() 
+	local CorIdUAI = self.mobileSession:SendRPC("UnregisterAppInterface",{}) 
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {appID = self.applications["SyncProxyTester"], unexpectedDisconnect = false})
+	EXPECT_RESPONSE(CorIdUAI, { success = true, resultCode = "SUCCESS"})
+	:Timeout(2000) 
+end
+
+function Test:Register_app()
+	commonTestCases:DelayedExp(3000)
+	self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+	self.mobileSession:StartService(7)
+	:Do(function()
+		local correlationId = self.mobileSession:SendRPC("RegisterAppInterface", config.application1.registerAppInterfaceParams)
+		EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered")
+		:Do(function(_,data)
+			self.HMIAppID = data.params.application.appID
+		end)
+		self.mobileSession:ExpectResponse(correlationId, { success = true, resultCode = "SUCCESS" })
+		self.mobileSession:ExpectNotification("OnHMIStatus", {hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+	end)
+end
+
+function Test:Activate_app()
+	local RequestId = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = self.HMIAppID})
+	EXPECT_HMIRESPONSE(RequestId)
+	EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+end
+
+commonSteps:DeactivateAppToNoneHmiLevel()
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+commonFunctions:userPrint(34, "Test is intended to check that isPermissionsConsentNeeded:true for app permissions that require consent")
+
+function Test:PTU_with_app_permissions_require_consent()
+	local RequestIdGetURLS = self.hmiConnection:SendRequest("SDL.GetURLS", { service = 7 })
+	EXPECT_HMIRESPONSE(RequestIdGetURLS,{result = {code = 0, method = "SDL.GetURLS", urls = {{url = "http://policies.telematics.ford.com/api/policies"}}}})
+	:Do(function(_,data)
+		self.hmiConnection:SendNotification("BasicCommunication.OnSystemRequest",
+			{
+				requestType = "PROPRIETARY",
+				fileName = "filename"
+			}
+		)
+		EXPECT_NOTIFICATION("OnSystemRequest", { requestType = "PROPRIETARY" })
+		:Do(function(_,data)
+				local CorIdSystemRequest = self.mobileSession:SendRPC("SystemRequest",
+					{
+						fileName = "PolicyTableUpdate",
+						requestType = "PROPRIETARY"
+					}, "files/PTU_NewPermissionsForUserConsent.json")
+				local systemRequestId
+				EXPECT_HMICALL("BasicCommunication.SystemRequest")
+				:Do(function(_,data)
```

Comment:
@aarotar in this case you have shadowed value of argument "data" of parent function. Please avoid shadowing arguments, use a new name for this argument.